### PR TITLE
[13/n] [dropshot_endpoint] support channels in API traits

### PR DESCRIPTION
**Pull request by sunshowers**
Building on the work in the last few PRs, this is relatively straightforward.
The bulk of the PR is simply tests for bad channels -- we now have mirrored
tests for all four combinations of (macro kind, endpoint kind).

### CODE REVIEW

**Comment by sunshowers** [comment]
@ahl whenever you're back a review on this would be really appreciated. It is mostly tests with ~500 lines of real code. thanks!

File: dropshot_endpoint/src/api_trait.rs [hunk 1]
```diff
@@ -1182,22 +1251,165 @@ impl<'ast> ApiEndpoint<'ast> {
...
+                    "endpoint `{name_str}` must be of the form \
+                     #[channel {{ protocol = WEBSOCKETS, path = \"/path\", ... }}]"
+                ),
+            ));
+            return None;
+        }
+    };
+
+    // TODO: Switch to from_tokenstream_spanned once
+    // https://github.com/oxidecomputer/serde_tokenstream/pull/194 is available.
```
Comment (ahl) [inline dropshot_endpoint/src/api_trait.rs:1275]: I think it is, right?

**Review by ahl — APPROVED** [review]
I didn't look at this very closely -- I kind of figure the tests speak for themselves. If there are parts that you think require a more fine-grained review, please let me know

File: dropshot_endpoint/src/api_trait.rs (hunk 1 above)
Comment (sunshowers) [reply dropshot_endpoint/src/api_trait.rs:1275]: Yeah, I missed removing this. Will do, thanks!

**Pull request merged**